src/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -970,6 +986,229 @@ public with sharing class CRLP_RollupUI_SVC {
+            }
+        }
+
+        /*******************************************************************************************************
+         * @description If the archive bridge needs to be notified about rollup type change, compare rollups.
+         */
+        private void checkIfChanged(Rollup__mdt rlp) {
+            if (!isRollupStateEnabled || hasChanged || isNew) {
+                return;
+            }
+
+            if (cmtRollup.recordName == rlp.DeveloperName) {
+                hasChanged = cmtRollup.isDifferent(rlp);
+
+                if (hasChanged) {
+                    oldRollupType = new CRLP_RollupCMT.Rollup(rlp).getRollupType();
+                }
+            }
+        }
+
+        /*******************************************************************************************************
+         * @description Determines if the rollup should be marked as stale in the archive bridge package
+         */
+        public Boolean shouldSendChangeEvent() {//or shouldSendChangeNotification?
+            return isRollupStateEnabled && (hasChanged || isNew);
+        }
+
+        /*******************************************************************************************************
+         * @description Builds a map of rollups that have changed
+         * @return Map<String, Object> Map of rollup types by the Callable API action
+         */
+        public Map<String, Object> getParams() {
+            if (oldRollupType == null) {
+                oldRollupType = rollupType;
+            }
+
+            Set<String> rollupTypes = new Set<String> { rollupType.name(), oldRollupType.name() };
+
+            return new Map<String, Object> {
+                CRLP_ApiService.PARAM_ROLLUP_TYPES => rollupTypes
+            };
+        }
+    }
+
+    /******************************************************************************************************************
+    * @description Utility class for Filter Group processing
+    */
+    public class FilterGroupUtil {
+
+        private Map<Id, List<Filter_Rule__mdt>> rulesByGroupId {
+            get {
+                if (rulesByGroupId == null) {
+                    rulesByGroupId = CMT_FilterRule_SEL.getFilterRulesByGroup();
+                }
+                return rulesByGroupId;
+            }
+            set;
+        }
+
+        private Boolean isRollupStateEnabled {
+            get {
+                if (isRollupStateEnabled == null) {
+                    isRollupStateEnabled = new CRLP_ApiService().isRollupStateEnabled();
+                }
+                return isRollupStateEnabled;
+            }
+            set;
+        }
+
+        /******************************************************************************************************************
+        * @description Determines if the archive bridge rollup state is enabled.
+        */
+        public Boolean isRollupStateEnabled() {
```

Comment:
"Rollup State" is a concept that, like "Stale", is unique to the DSO Bridge. From the viewpoint of NPSP, it needs to know if there is some extension installed that needs to know when a Rollup or FilterRule is changed. How about something like `isRollupChangeTriggerEnabled' or `isCrlpChangeTriggerEnabled`, at least within this class. Have to look at the usage of the method in the parent branch to see if a name change is warranted there.